mantidproject/mantid
# Added more tests for new elemental analysis gui

**Description of work.**
This pull request adds extra tests for the new elemental analysis gui, updates the rebin input dialog to inform users that rebininig will create a new workspace and updating grouping tab so as to produce a warning popup when no parameters are given.

**Report to:** Adrian Hillier


**To test:**
1. Before starting Mantid go to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L24 , add Muon/Elemental_Analysis_2.py to the end of the line and build Mantid.

2. Download the test files from PR #29958 and add to User Directories.

3. Open the elemental analysis 2 GUI and load 9997.
4. In the rebin column select fixed and leave empty, click on okay and warning box should come up with the message "Rebin parameters not given"
5. In the rebin column select variable and leave empty, click on okay and warning box should come up with the message "Rebin parameters not given"


Fixes #30214 and #31175.





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
